internal/helm/controller/controller.go

```diff
@@ -97,6 +98,9 @@ func Add(mgr manager.Manager, options WatchOptions) error {
 // watchDependentResources adds a release hook function to the HelmOperatorReconciler
 // that adds watches for resources in released Helm charts.
 func watchDependentResources(mgr manager.Manager, r *HelmOperatorReconciler, c controller.Controller) {
+	// using predefined functions for filtering events
+	dependentPredicate := predicate.DependentPredicateFuncs()
```

Comment:
Can we have a little more context as to why this is getting added? Given the context is "dependency updates", this reads to me like "not a dependency update".

Reply:
If it's not caused/related to a dependency change, then it probably doesn't need to be in this PR.

Reply:
I'm either coding this wrong, or operator-lib no longer has what I need, without pulling this code into the project this code would not build, so this (or something equivalent) has to be part of this PR.

Reply:
It seems I was coding this wrong, and have removed this predicate file, and went back to using things from operator-lib.